### PR TITLE
Removed unnecessary references to params to avoid inheritance issues

### PR DESCRIPTION
**Pull request by t0mmyt**
Pretty much that.  None of the sub classes should be called directly so $unbound:: is already in scope.  By calling or referring to unbound::params values defined when calling the unbound class are forgotten about. (e.g. an alternative $config_file).

### CODE REVIEW

File: manifests/remote.pp [hunk 1]
```diff
@@ -6,15 +6,15 @@
...
-  $group             = $unbound::params::group,
-  $confdir           = $unbound::params::confdir,
-) inherits unbound::params {
+  $server_key_file   = "${unbound::confdir}/unbound_server.key",
+  $server_cert_file  = "${unbound::confdir}/unbound_server.pem",
+  $control_key_file  = "${unbound::confdir}/unbound_control.key",
+  $control_cert_file = "${unbound::confdir}/unbound_control.pem",
+  $group             = $unbound::group,
+  $confdir           = $unbound::confdir,
+) inherits unbound {
```
Comment (igalic) [inline manifests/remote.pp]: i'm a bit unhappy about **this** change.
normally, sub-classes should inherit from *only* from ::params, and only if they are top-level classes.
otherwise, they shouldn't have no parameter themselves (and be called from init.pp)

**Comment by igalic** [comment]
the rest looks good imo, :+1:

File: manifests/remote.pp (hunk 1 above)
Comment (t0mmyt) [inline manifests/remote.pp]: That makes sense, I think I got a little trigger happy on my search and replace.

Comment (xaque208) [inline manifests/remote.pp]: I'd agree with @igalic here.  Inheritance isn't really recommended, and just about the only place its deemed acceptable is on the params class.  Mostly just stylistic.

File: manifests/forward.pp [hunk 2]
```diff
@@ -7,9 +7,7 @@
   $zone = $name,
 ) {
 
-  include unbound::params
-
-  $config_file = $unbound::params::config_file
+  $config_file = $unbound::config_file
```
Comment (xaque208) [inline manifests/forward.pp]: This is an out of scope variable unless we `include unbound`.

**Comment by xaque208** [comment]
@t0mmyt Are you able to address the comments here?

**Comment by igalic** [comment]
@t0mmyt ping

**Comment by xaque208** [comment]
Closing PR due to lack of response.  If someone else wants to pick this work up, PRs welcome.
